sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -131,6 +138,11 @@ void *alignedAllocInternal(size_t Alignment, size_t Size,
                            const context_impl *CtxImpl,
                            const device_impl *DevImpl, alloc Kind,
                            const property_list &PropList) {
+  if ((Kind == alloc::device || Kind == alloc::shared) &&
+      !DevImpl->has(sycl::aspect::usm_device_allocations)) {
+    throw sycl::exception(sycl::errc::feature_not_supported,
+                          "Device does not support Unified Shared Memory!");
+  }
```

Comment:
It seems that there is the same mistake at this part.
We have three aspects:
usm_host_allocations
usm_device_allocations
usm_shared_allocations

I think we need to check corresponding aspect depending on Kind argument and throw if there is a lack of support.